app/user_api/User.py

```diff
@@ -29,29 +29,61 @@ def __init__(
         self.hasCompletedOnboarding = hasCompletedOnboarding
 
     def get(db_ref, id):
+        """
+        Fetches a specified user from a database.
+
+        Args:
+            db_ref: A `CollectionReference` to the users table from Firestore
+            id: A string of the user email to fetch
+
+        Returns:
+            User: specified by `id`
+
+        Raises:
+            UserNotFoundError: if the target user does not exist
+        """
         target_data = db_ref.document(id).get()
         if not target_data.exists:
             raise UserNotFoundError(id)
         return User.from_dict(db_ref, target_data.to_dict())
 
     def from_dict(db_ref, dict):
```

Comment:
Not really part of this review, but the close coupling of data entities with the database is an anti-pattern. I shouldn't need a database reference to parse an object from a database object. This should be refactored to move any database logic out of the class and into service methods. I think we need to re-think the base class we currently have.

Reply:
Would it be better to just initialize the database reference inside the data class so that it doesn't need to be given a database reference at all?

Reply:
The database reference should be kept separate from the data class completely. It should be passed to the classes functions that need it. E.g. If you have a service method that pulls from the database it can create a database connection and take in a data object.

Reply:
So with all the database logic in the service class, the service class should use the data to initialize the data class? In this case, is there much use for a data class? Wouldn't it be easier to just keep them in a Python dictionary since all we'd be doing with it is fetching properties from it anyway?

Reply:
It's an abstraction layer between the data in the database and whatever functions you have in the app. If all you're doing is returning the data straight away, not really. But in theory your data object has extra functions that you'll use throughout the backend more than just pulling data from the database.